jobqueue/src/main/java/com/birbit/android/jobqueue/persistentQueue/sqlite/FileStorage.java

```diff
@@ -0,0 +1,81 @@
+package com.birbit.android.jobqueue.persistentQueue.sqlite;
+
+
+import android.content.Context;
+import android.support.annotation.Nullable;
+
+import com.birbit.android.jobqueue.log.JqLog;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Set;
+
+import okio.BufferedSink;
+import okio.BufferedSource;
+import okio.Okio;
+
+/**
+ * Provides a file based storage to keep jobs.
+ * This class is NOT thread safe and re-uses Buffers
+ */
+public class FileStorage {
+    private static final String EXT = ".jobs";
+    private File folder;
+    FileStorage(Context appContext, String id) {
+        this.folder = new File(appContext.getDir("com_birbit_jobqueue_jobs", Context.MODE_PRIVATE),
+                "files_" + id);
+        //noinspection ResultOfMethodCallIgnored
+        this.folder.mkdirs();
+    }
+
+    void delete(String id) {
+        File file = new File(folder, filename(id));
+        if (file.exists()) {
+            //noinspection ResultOfMethodCallIgnored
+            file.delete();
+        }
+    }
+
+    @Nullable
+    byte[] load(String id) throws IOException {
+        File file = new File(folder, filename(id));
+        if (file.exists() && file.canRead()) {
+            BufferedSource source = Okio.buffer(Okio.source(file));
+            return source.readByteArray();
+        }
+        return null;
+    }
+
+    void save(String id, byte[] data) throws IOException {
+        File file = new File(folder, filename(id));
+        BufferedSink sink = Okio.buffer(Okio.sink(file));
+        sink.write(data).flush();
+    }
+
+    private static String filename(String id) {
```

Comment:
Is it worth to change this method to return a `File` object? Or maybe create a new one that does that? I think all the time you invoke it you also create a `File`a round it